### PR TITLE
correccted resource location in azuredeploy.parameters.json

### PR DESCRIPTION
**Pull request by mddazure**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

### Changelog

*
*
*

### CODE REVIEW

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.parameters.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+          "description": "Username for the Virtual Machine."
+        }
+      },
+      "adminPassword": {
+        "value": "GEN-PASSWORD",
+        "metadata": {
+          "description": "Password for the Virtual Machine."
+        }
+      },
+      "windowsOSVersion": {
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.parameters.json]: remove this and use the defaultValue in the template

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 2]
```diff
@@ -0,0 +1,1518 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "adminUsername": {
+            "type": "string",
+            "defaultValue": "azfw",
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: remove defaultValue (increases surface area)

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,1518 @@
...
+                "[resourceId('Microsoft.Network/virtualNetworks', variables('virtualNetworks_azfw_on_prem_name'))]"
+            ],
+            "properties": {
+                "addressPrefix": "172.16.0.224/27",
+                 
+                "privateEndpointNetworkPolicies": "Enabled",
+                "privateLinkServiceNetworkPolicies": "Enabled"
+            }
+        },
+        {
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: how is this used?

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,1518 @@
...
+                    "networkInterfaces": [
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces', variables('networkInterfaces_onprem_name'))]"
+                        }
+                    ]
+                },
+                "diagnosticsProfile": {
+                    "bootDiagnostics": {
+                        "enabled": true,
+                        "storageUri": "[concat('https://', variables('storageAccounts_azfwdemodiag_name'), '.blob.core.windows.net/')]"
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#referencing-resource-properties) for how to handle storage account uris

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,1518 @@
...
+                    "networkInterfaces": [
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces', variables('networkInterfaces_spokevm1_name'))]"
+                        }
+                    ]
+                },
+                "diagnosticsProfile": {
+                    "bootDiagnostics": {
+                        "enabled": true,
+                        "storageUri": "[concat('https://', variables('storageAccounts_azfwdemodiag_name'), '.blob.core.windows.net/')]"
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#referencing-resource-properties) for how to handle storage account uris

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,1518 @@
...
+                    "networkInterfaces": [
+                        {
+                            "id": "[resourceId('Microsoft.Network/networkInterfaces', variables('networkInterfaces_spokevm2_name'))]"
+                        }
+                    ]
+                },
+                "diagnosticsProfile": {
+                    "bootDiagnostics": {
+                        "enabled": true,
+                        "storageUri": "[concat('https://', variables('storageAccounts_azfwdemodiag_name'), '.blob.core.windows.net/')]"
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: See [BP guide](https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md#referencing-resource-properties) for how to handle storage account uris

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 7]
```diff
@@ -0,0 +1,1518 @@
...
+            "type": "Microsoft.Network/bastionHosts",
+            "apiVersion": "2019-09-01",
+            "name": "[variables('bastionHosts_azfw_hub_bastion_name')]",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/publicIPAddresses', variables('publicIPAddresses_azfw_hub_ip_name'))]",
+                "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworks_azfw_hub_name'), 'AzureBastionSubnet')]"
+            ],
+            "properties": {
+                "dnsName": "bst-88091e28-bfb2-40d7-a372-17daafc8cbc1.bastion.azure.com",
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: remove this property

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 8]
```diff
@@ -0,0 +1,1518 @@
...
+            "type": "Microsoft.Network/bastionHosts",
+            "apiVersion": "2019-09-01",
+            "name": "[variables('bastionHosts_onprem_bastion_name')]",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/publicIPAddresses', variables('publicIPAddresses_azfw_on_prem_ip_name'))]",
+                "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworks_azfw_on_prem_name'), 'AzureBastionSubnet')]"
+            ],
+            "properties": {
+                "dnsName": "bst-bc2e730d-498d-4f4d-828c-92888981428a.bastion.azure.com",
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: remove this property

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 9]
```diff
@@ -0,0 +1,1518 @@
...
+            "type": "Microsoft.Network/bastionHosts",
+            "apiVersion": "2019-09-01",
+            "name": "[variables('bastionHosts_spoke1_bastion_name')]",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/publicIPAddresses', variables('publicIPAddresses_azfw_spoke1_ip_name'))]",
+                "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworks_azfw_spoke1_name'), 'AzureBastionSubnet')]"
+            ],
+            "properties": {
+                "dnsName": "bst-fd97b18a-2f11-4105-84f4-5dc4f4a912f1.bastion.azure.com",
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: remove this property

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 10]
```diff
@@ -0,0 +1,1518 @@
...
+            "type": "Microsoft.Network/bastionHosts",
+            "apiVersion": "2019-09-01",
+            "name": "[variables('bastionHosts_spoke2_bastion_name')]",
+            "location": "[parameters('location')]",
+            "dependsOn": [
+                "[resourceId('Microsoft.Network/publicIPAddresses', variables('publicIPAddresses_azfw_spoke2_ip_name'))]",
+                "[resourceId('Microsoft.Network/virtualNetworks/subnets', variables('virtualNetworks_azfw_spoke2_name'), 'AzureBastionSubnet')]"
+            ],
+            "properties": {
+                "dnsName": "bst-dd620c26-35a0-4473-ae4b-64c84600a8c8.bastion.azure.com",
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: remove this property

**Comment by mddazure** [comment]
Incorporated changes requested, also fixed shared key for S2S VPN. Automtaed test for Public passes. Deployment test to Gov fails because come resource types used are not available in Gov yet.

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 11]
```diff
@@ -84,7 +80,8 @@
         "routeTables_azfw_spokes_via_fw_name": "azfw-spokes-via-fw",
     
         "virtualNetworkGateways_azfw_hub_gw_name": "azfw-hub-gw",
-        "virtualNetworkGateways_azfw_on_prem_gw_name": "azfw-on-prem-gw"    
+        "virtualNetworkGateways_azfw_on_prem_gw_name": "azfw-on-prem-gw",
+        "virtualNetworkGateways_shared_key":"[uniquestring(subscription().subscriptionId)]"
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: what's the requirement for uniqueness here?  This will allow for only one per subscription...

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json [hunk 12]
```diff
@@ -4,16 +4,12 @@
...
         "adminUsername": {
             "type": "string",
-            "defaultValue": "azfw",
             "metadata": {
               "description": "Username for the Virtual Machine."
             }
           },
           "adminPassword": {
-            "type": "securestring",
-            "metadata": {
```
Comment (bmoore-msft) [inline 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: revert this change (add the metadata property back)

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.parameters.json (hunk 1 above)
Comment (bmoore-msft) [reply 101-azurefirewall-hub-spoke-demo/azuredeploy.parameters.json]: this is still needed

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json (hunk 11 above)
Comment (mddazure) [reply 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: Wanted to create a unique key for S2S connection. Subscr id as a seed may not be the best idea, you're right. Changed it to use deployment().name as the seed.

File: 101-azurefirewall-hub-spoke-demo/azuredeploy.json (hunk 12 above)
Comment (mddazure) [reply 101-azurefirewall-hub-spoke-demo/azuredeploy.json]: Added metadata description back in the template and removed from the parameters file.
